src/cpp/rtps/builtin/discovery/participant/PDP.cpp

```diff
@@ -695,6 +695,8 @@ ReaderProxyData* PDP::addReaderProxyData(
                 RTPSParticipantListener* listener = mp_RTPSParticipant->getListener();
                 if (listener)
                 {
+                    //Unlock before calling listeners
+                    lockPDP.unlock();
```

Comment:
I know this was the main purpose of this PR, but we cannot unlock here, as `ReaderDiscoveryInfo` just holds a const ref to `ReaderProxyData` and not a copy.

The same applies to `WriterDiscoveryInfo` and `WriterProxyData`.

So I am afraid we should revert all changes on this file.

Reply:
So true.
reverted.